### PR TITLE
CI: PGO Profile generation

### PR DESCRIPTION
**Pull request by sammacbeth**
Adds a new CI job to run profile generation for PGO builds.

### CODE REVIEW

File: ci/profileserver_linux.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#! /bin/bash -vex
```
Comment (chrmod) [inline ci/profileserver_linux.sh:1]: how about calling those files `linux_*` and `win_*` so all platform specific files are prefixed?

File: ci.pgo.Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,128 @@
+
```
Comment (chrmod) [inline ci.pgo.Jenkinsfile:1]: How about calling this file `ci.pgo-profile.Jenkinsfile` or something like that?

File: ci/profileserver_linux.sh (hunk 1 above)
Comment (sammacbeth) [reply ci/profileserver_linux.sh:1]: Sure. This was following the convention of our `sign_{mac,win}` scripts, so we should probably also change those too.

Comment (chrmod) [reply ci/profileserver_linux.sh]: yep, will do that

**Pull request merged**